File: frontend/src/types/strategy.ts
export interface StrategyStep {
  id: string
  title: string
  description?: string
  frequency?: string
  xp_reward: number
}

export interface Strategy {
  id: string
  domain: string
  name: string
  description?: string
  difficulty: string
  estimated_impact: number
  steps: StrategyStep[]
}

export interface ActiveStrategy {
  id: string
  strategy_id: string
  domain: string
  name: string
  description?: string
  started_at?: string
  adherence_score: number
  steps: Array<{ id: string; title: string; frequency?: string; xp_reward: number }>
}

File: backend/app/schemas/strategy.py
from typing import Optional
from pydantic import BaseModel


class StrategyStepBase(BaseModel):
    title: str
    description: Optional[str] = None
    frequency: Optional[str] = None
    xp_reward: float = 0


class StrategyStepResponse(StrategyStepBase):
    id: str


class StrategyBase(BaseModel):
    domain: str
    name: str
    description: Optional[str] = None
    difficulty: str = "medium"
    estimated_impact: float = 0


class StrategyResponse(StrategyBase):
    id: str
    created_at: Optional[str] = None
    steps: list[StrategyStepResponse] = []


class StrategyRecommended(StrategyResponse):
    pass


class UserStrategyActivate(BaseModel):
    strategy_id: str


class ActiveStrategyResponse(BaseModel):
    id: str
    strategy_id: str
    domain: str
    name: str
    description: Optional[str] = None
    started_at: Optional[str] = None
    adherence_score: float
    steps: list[dict] = []
